capstone-project/src/main/angular-webapp/src/app/info-window/info-window.component.spec.ts

```diff
@@ -22,47 +29,121 @@ describe('InfoWindowComponent', () => {
   });
 
+  it('Should initialize blobKeyValue with the original blob key', () => {
+
+    component.originalBlobKey = "originalKey";
+    component.ngOnInit();
+    expect(component.getBlobKeyValue()).toBe("originalKey");
+  });
+
   it('Should emit on submit', () => {
 
+    component.type = MarkerMode.CREATE;
+    fixture.detectChanges();
     spyOn(component.submitEvent, 'emit');
+    
     // Trigger the submit event
-    component.submit("animal", "description", "reporter");
+    fixture.debugElement.nativeElement.querySelector('#submitButton').click();
 
-    fixture.detectChanges();
-
-    expect(component.submitEvent.emit).toHaveBeenCalledWith({animal: "animal", description: "description", reporter: "reporter", blobKey: component.getBlobKeyValue() });
+    expect(component.submitEvent.emit).toHaveBeenCalled();
   });
 
   it('Should emit on delete', () => {
 
+    component.type = MarkerMode.USER_VIEW;
+    fixture.detectChanges();
     spyOn(component.deleteEvent, 'emit');
-    // Trigger the delete event
-    component.delete();
 
-    fixture.detectChanges();
+    // Trigger the delete event
+    fixture.debugElement.nativeElement.querySelector('#deleteButton').click();
 
     expect(component.deleteEvent.emit).toHaveBeenCalled();
   });
 
   it('Should emit on update', () => {
 
+    component.type = MarkerMode.USER_VIEW;
+    fixture.detectChanges();
     spyOn(component.updateEvent, 'emit');
-    // trigger the update event
-    component.update();
 
-    fixture.detectChanges();
+    // Trigger the update event
+    fixture.debugElement.nativeElement.querySelector('#updateButton').click();
 
     expect(component.updateEvent.emit).toHaveBeenCalled();
   });
 
   it('Should emit on cancel', () => {
 
+    component.type = MarkerMode.UPDATE;
+    fixture.detectChanges();
     spyOn(component.cancelEvent, 'emit');
-    // trigger the cancel event
-    component.cancel();
 
-    fixture.detectChanges();
+    // Trigger the cancel event
+    fixture.debugElement.nativeElement.querySelector('#cancelButton').click();
 
     expect(component.cancelEvent.emit).toHaveBeenCalled();
   });
+
+  it('Should post the file that was uploaded', async () => {
+
+    // Create an instance of type 'create' so that it has the 'file-name' element
+    component.type = MarkerMode.CREATE;
+    fixture.detectChanges();
+
+    // Create a file to post
+    const fileList: FileList = {
+      length: 1,
+      item(index: number): File {
+        return fileList[index];
+      }
+    };
+    fileList[0] = new File(["file data"], "file.txt");
+
+    // Mock the document.getElementById function to avoid accessing the tests page's document instead of the component's document
+    const dummyElement = fixture.debugElement.nativeElement.querySelector('#file-name');
+    spyOn(document, "getElementById").and.returnValue(dummyElement);
+
+    await component.postFile(fileList)
+
+    expect(component.getBlobKeyValue()).toBe("blobKey");
```

Comment:
This's a magic string, if we change MockInterceptor it will be hard to detect why this fails. 

You can either: 
1. Allow setting the expected outcome of MockInterceptor
2. have "blobKey" as a constant in MockInterceptor and use it here.